src/on-track-url-change.js

```diff
@@ -0,0 +1,39 @@
+const functions = require('firebase-functions')
+const admin = require('firebase-admin');
+const getYoutubeId = require('./utils/youtube-url-to-id.js')
+
+/*
+	To test the execution of this function
+	you can run `yarn firebase:shell`
+	and use the following mockup call
+	
+	onTrackUrlChange({before: 'youtu.be/xIaco5AQrUQ', after: 'https://www.youtube.com/watch?v=OkR7UNnQU6c' })
+*/
+
+const onTracksChange = (snapshot, context) => {
+	// console.log('snapshot: ', snapshot)
+	// console.log('context: ', context)
+	
+	let db = admin.database();
+
+	let newUrl = snapshot.after.val()
+	let providerId = getYoutubeId(newUrl)
+	let providerName = 'youtube'
+	
+	// abort if no ID in provider URL
+	if(!providerId) {
+		return false
+	}
+	
+	const ref = db.ref(`/providerTracks/${providerName}:${providerId}/tracks/${context.params.trackId}`);
```

Comment:
How about splitting `providerName` and `providerId` with a slash as well? Instead of the `:`.  So you don't have to parse the string to figure out which provider it is. Easier to query with firebase I assume.

Reply:
Right now the DB looks like this
```
- providerTracks
- - youtube:OkR7UNnQU6c
```
So if you want to query it with ember your just do `store.find('providerTrack', youtube:OkR7UNnQU6c)`.

If you separate the provider with a slash how would you query the `providerTrack` that has the youtube id `OkR7UNnQU6c`?

Reply:
True. Not sure how to query that either. Maybe different models. Each extend `providerTrack` so the adapter can specify the endpoints: `this.store.find('youtube-track', id)`? Ala what we do on the Explorer for Discogs.

---

If we have `discogs:123` and `youtube:456`. How do you query all Discogs tracks?

Another idea is to move the provider to to a property? `{id: '123', provider: 'youtube'}`. Then it's easy to filter and query. Trying to think of the pros and cons..

Reply:
It seems like there is a misunderstanding on the point of this new model.

I will try to make a summary to explain the idea better.